src/core/style/style_parameter.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2023 MapLibre contributors
+
+// SPDX-License-Identifier: BSD-2-Clause
+
+#ifndef QMAPLIBRE_STYLE_PARAMETER_H
+#define QMAPLIBRE_STYLE_PARAMETER_H
+
+#include <QMapLibre/Export>
+
+#include <QtCore/QObject>
+#include <QtCore/QString>
+#include <QtCore/QVariantMap>
```

Comment:
Can wrap these in `#ifndef Q_MOC_RUN` to save on build time (same for other classes below that will be moc'd)

Reply:
Let me follow-up on this later as I never used that before and need to read a bit about it first.

Opened #83.